hrqb/base/task.py

```diff
@@ -1,69 +1,112 @@
 """hrqb.base.task"""
 
+import os
+from abc import abstractmethod
+from typing import Literal
+
 import luigi  # type: ignore[import-untyped]
 import pandas as pd
 
 from hrqb.base import PandasPickleTarget, QuickbaseTableTarget
+from hrqb.config import Config
 from hrqb.utils import today_date
 from hrqb.utils.quickbase import QBClient
 
 
 class HRQBTask(luigi.Task):
     """Base Task class for all HRQB Tasks."""
 
-    path = luigi.Parameter()
-    table_name = luigi.Parameter()
+    pipeline = luigi.Parameter()
+    stage: Literal["Extract", "Transform", "Load"] = luigi.Parameter()
+    table_name = luigi.OptionalStrParameter(default=None)
+
+    @property
+    def path(self) -> str:
+        """Dynamically generate path for HRQBTask Targets.
+
+        The pattern is:
+            Targets Dir + Pipeline Name + Pipeline Stage + Task Name + File Extension
+
+        Examples:
+            output/LookupTablesPipeline__extract__JobData.pickle
+            output/LookupTablesPipeline__transform__UniqueJobTitles.pickle
+            output/LookupTablesPipeline__load__UpsertJobTitles.pickle
+        """
+        filename = (
+            "__".join(  # noqa: FLY002
```

Comment:
The double underscore `__` is used to support splitting on this filename if needed, where a single underscore _could_ showup in parts of the name.  Splitting on the double underscore would give the:
- pipeline name
- pipeline stage
- task name